docs/proposals/control-plane-config.md

```diff
@@ -17,3 +17,98 @@ option that we will support is log level.
 ## Non-Goals
 
 This proposal is *not* defining a way to dynamically configure the data plane.
+
+## Introduction
+
+The NKG control plane will evolve to have various user-configurable options. These could include, but are not
+limited to, log level, tracing, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NKG. The first option that we will allow users to configure is the
+log level. The easiest and most intuitive way to implement a Kubernetes-native API is through a CRD.
+
+In this doc, the term "user" will refer to the cluster operator (the person who installs and manages NKG). The
+cluster operator owns this CRD resource.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NGINXControlConfig
+metadata:
+    name: nkg-config
+    namespace: nginx-gateway
+spec:
+    log:
+        level: info
+    ...
+status:
+    ...
+```
+
+- The CRD would be Namespace-scoped, living in the same Namespace as the controller that it applies to.
+- CRD is initialized and created when NKG is deployed.
+- NKG references the name of this CRD via CLI arg, and only watches this CRD. If the resource doesn't exist,
+then an error is logged and event created, and default values are used.
+- If user deletes resource, NKG logs an error and creates an event. NKG will revert to default values.
+
+This resource won't be referenced in the `parametersRef` of the GatewayClass, reserving that option for a data
+plane CRD. The control plane may end up supporting multiple GatewayClasses, so linking the control CRD to a
+GatewayClass wouldn't make sense. Referencing the CRD via a CLI argument ensures we only support one instance of
+the CRD per control plane.
+
+For discussion with team:
+
+- kind name
+- default resource name
```

Comment:
Would it make sense to use `NginxControlConfig` instead of `NGINXControlConfig`?
While in the docs we use NGINX most of the time, but in the code we use `Nginx` like `ListenerMessageFailedNginxReload` or `NginxFileMgr`. NGINX is just the result of marketing and writing guidelines.  If we use `NginxControlConfig`, we will be able use `NginxControlConfig` (it will be exactly the same as the kind) as the name of the go struct too, so it would be consistent with the existing code.

Reply:
Envoy Gateway uses the Kinds `EnvoyGateway` for the control config and `EnvoyProxy` for the dataplane config: https://github.com/envoyproxy/gateway/blob/07297df653e96751b5d1adffdc06c00b92aa431a/docs/v0.5.0/api/config_types.md#envoygateway

What do you guys think of following that pattern? `NginxGateway` and `NginxProxy`? 

Default resource name ideas: `config`, `ctrl-config`, `nginx-ctrl-config`, `nginx-gateway-config`.

Reply:
Not bad ideas...`Config` is likely redundant since all CRDs are config...

Does `NginxGateway` seem clear enough that it's for control plane config? I guess as long as our docs explain what this thing is, the name doesn't have to be *too* specific.

I think I like `nginx-gateway-config` the best out of the resource names.

Reply:
I think it will be pretty clear what the resource is for. What are you thinking about using for the cli flag name? Should we include that in this doc?

Reply:
> What do you guys think of following that pattern? NginxGateway and NginxProxy?

A few points:

There is a GEP https://gateway-api.sigs.k8s.io/geps/gep-1867/ that will allow references a CR from the Gateway resource. I wonder if `NginxGateway` will become confusing considering that?

If we entertain a possibility of supporting additional data planes, will `NginxGateway` and `NginxProxy`, `SomethingProxy` work?

Reply:
For CLI arg, `--control-config-name`.

Reply:
> For CLI arg, --control-config-name.

If we go with `NginxGateway` should we change this to `--nginx-gateway-config-name`?